java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestSearchUtils.java

```diff
@@ -382,7 +382,38 @@ private static void findTestItemsInTypeBinding(ITypeBinding typeBinding, JavaTes
         for (final ITypeBinding childTypeBinding : typeBinding.getDeclaredTypes()) {
             findTestItemsInTypeBinding(childTypeBinding, classItem, searchers, monitor);
         }
+    }
 
+    /**
+     * Given a file Uri, resolve its belonging project and package node in the test explorer, this is
+     * used to find the test item node when a test file is changed
+     * @param arguments A list containing a uri of the file
+     * @param monitor Progress monitor
+     * @throws JavaModelException
+     */
+    public static List<JavaTestItem> resolvePath(List<Object> arguments, IProgressMonitor monitor)
+            throws JavaModelException {
+        final List<JavaTestItem> result = new LinkedList<>();
+        final String uriString = (String) arguments.get(0);
+        if (JavaCore.isJavaLikeFileName(uriString)) {
+            final ICompilationUnit unit = JDTUtils.resolveCompilationUnit(uriString);
+            if (unit == null) {
+                return Collections.emptyList();
+            }
+            final IJavaProject project = unit.getJavaProject();
+            if (project == null) {
+                return Collections.emptyList();
+            }
+            result.add(TestItemUtils.constructJavaTestItem(project, TestLevel.PROJECT, TestKind.None));
+
+            final IPackageFragment packageFragment = (IPackageFragment) unit.getParent();
+            if (packageFragment == null || !(packageFragment instanceof IPackageFragment)) {
+                return Collections.emptyList();
```

Comment:
what about the project level item you just added into `result` above?

Reply:
Just forget about it since only one project node is not enough to resolve the entire path from root to leaf